runtime/afl.c

```diff
@@ -13,26 +13,34 @@
 /**************************************************************************/
 
 /* Runtime support for afl-fuzz */
+
+#define CAML_INTERNALS
+
 #include "caml/config.h"
+#include "caml/memory.h"
+#include "caml/mlvalues.h"
 
 /* Values used by the instrumentation logic (see cmmgen.ml) */
-static unsigned char afl_area_initial[1 << 16];
-unsigned char* caml_afl_area_ptr = afl_area_initial;
+
+#define INITIAL_AFL_AREA_SIZE (1 << 16)
+unsigned char * caml_afl_area_ptr = NULL;
 uintnat caml_afl_prev_loc;
 
 #if !defined(HAS_SYS_SHM_H) || !defined(HAS_SHMAT)
 
-#include "caml/mlvalues.h"
-#include "caml/domain.h"
-
-CAMLprim value caml_reset_afl_instrumentation(value full)
+CAMLexport value caml_setup_afl(value unit)
 {
+  /* AFL is not supported, but we still need to allocate space for the bitmap
+       (the instrumented OCaml code will write into it). */
+  if (caml_afl_area_ptr == NULL) {
+    caml_afl_area_ptr = caml_stat_alloc(INITIAL_AFL_AREA_SIZE);
+    memset(caml_afl_area_ptr, 0, INITIAL_AFL_AREA_SIZE);
```

Comment:
Just an aside while looking over this PR:  it might be useful to have a `caml_stat_calloc` call in the runtime that uses `calloc()` instead of `malloc(); memset()`.  We only have `caml_stat_calloc_noexc` which wouldn't be much of a saving here, and it just uses `memset()` under the hood anyway.